pkg/pod/idHandler.go

```diff
@@ -1,22 +1,24 @@
 package pod
 
 import (
-	"log"
+	"log/slog"
 
 	v1 "k8s.io/api/core/v1"
 )
 
-type idHandler struct{}
+type idHandler struct {
+	logger *slog.Logger
+}
 
-func NewIDHandler() idHandler {
-	return idHandler{}
+func NewIDHandler(logger *slog.Logger) idHandler {
+	return idHandler{logger}
 }
 
 func (h idHandler) Supports() string {
 	return "im-id"
 }
 
 func (h idHandler) Handle(pod v1.Pod) error {
-	log.Printf("Id handler invoked: %s", pod.Name)
+	h.logger.Info("Id handler invoked on", "pod", pod.Name)
```

Comment:
```suggestion
	h.logger.Info("Id handler invoked", "pod", pod.Name)
```

What is this handler for? looks like a noop, do we need it then?